airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresSqlGeneratorIntegrationTest.java

```diff
@@ -131,4 +133,34 @@ public void testCascadeDrop() throws Exception {
     assertDoesNotThrow(() -> getDestinationHandler().execute(generator.overwriteFinalTable(getIncrementalDedupStream().getId(), "_soft_reset")));
   }
 
+  /**
+   * Verify that when cascadeDrop is disabled, an error caused by dropping a table with dependencies
+   * results in a configuration error with the correct message.
+   */
+  @Test
+  public void testCascadeDropDisabled() throws Exception {
+    // Create a sql generator with cascadeDrop=false (this simulates what the framework passes from the
+    // config).
+    final PostgresSqlGenerator generator = new PostgresSqlGenerator(new PostgresSQLNameTransformer(), false);
+
+    // Create a table in the test namespace with a default name.
+    getDestinationHandler().execute(generator.createTable(getIncrementalAppendStream(), "", false));
+
+    // Create a view in the test namespace that selects from the test table.
+    // (Ie, emulate a client action that creates some dependency on the table.)
+    database.execute(createView(quotedName(getIncrementalAppendStream().getId().getFinalNamespace(), "example_view"))
```

Comment:
not a problem with your code, just fyi - at some point, we'll want to convert this class to kotlin, and then we can switch all the jooq code gen to kotlin string templating. Not really worth doing that as part of this PR though.